addons/main/script_macros_common.hpp

```diff
@@ -796,6 +796,19 @@ Macro: GVARMAIN()
     #define PREPMAIN(var1) ['PATHTO_SYS(PREFIX,COMPONENT_F,DOUBLES(fnc,var1))', 'TRIPLES(PREFIX,fnc,var1)'] call SLX_XEH_COMPILE_NEW
 #endif
 
+#ifdef RECOMPILE
+    #undef RECOMPILE
+    #define RECOMPILE recompile = 1
+#else
+    #undef RECOMPILE
+    #define RECOMPILE recompile = 0
+#endif
+
+#define PATHTO_FNC(func) class func {\
+    file = QUOTE(PATHTOF(DOUBLES(fnc,func).sqf));\ // \MAINPREFIX\PREFIX\SUBPREFIX\COMPONENT\fnc_func.sqf
```

Comment:
Should use a function (macro) header for explaining the usage instead of having in code comments:
```
/* -------------------------------------------
Macro: PATHTO_FNC()
 
Description:
    Defines a function inside CfgFunctions.
 
    Full file path in addons: '\MAINPREFIX\PREFIX\SUBPREFIX\COMPONENT\fnc_<FNC>.sqf'
    Define 'RECOMPILE' to enable recompiling for that function.

Parameters:
    FUNCTION NAME - Name of the function, unquoted <STRING>
 
Examples:
    (begin example)
        // file name: fnc_addPerFrameHandler.sqf
        class CfgFunctions {
            class CBA {
                class Misc {
                    PATHTO_FNC(addPerFrameHandler);
                };
            };
        };
        // -> CBA_fnc_addPerFrameHandler
    (end)
 
Author:
    dixon13
 ------------------------------------------- */
```

Or something like that.